services/web/server/src/simcore_service_webserver/catalog_api_handlers.py

```diff
@@ -0,0 +1,402 @@
+
+VX = f"/{api_version_prefix}"
+
+routes = RouteTableDef()
+
+
+@dataclass
+class _RequestContext:
+    app: web.Application
+    user_id: int
+    product_name: str
+
+    @classmethod
+    def create(cls, request: Request) -> "_RequestContext":
+        return cls(
+            app=request.app,
+            user_id=request[RQT_USERID_KEY],
+            product_name=request[RQ_PRODUCT_KEY],
+        )
+
+
+@contextmanager
+def parameters_validation(request: web.Request):
+    """Context manager to wrap match, parsing and validation of
+    request parameters (both path and queries)
+    """
+    try:
+        try:
+            context = _RequestContext.create(request)
+        except KeyError:
+            raise web.HTTPBadRequest(reason="Invalid headers")
+
+        yield context
+        #
+        # wraps match, parse and validate
+        # For instance
+        #   service_key: ServiceKey = request.match_info["service_key"]
+        #   from_service_version: ServiceVersion = request.query["fromVersion"]
+        #
+    except ValidationError as exc:
+        raise web.HTTPUnprocessableEntity(
+            text=jsonify({"error": exc.errors()}), content_type="application/json"
+        ) from exc
+
+    except KeyError as err:
+        raise web.HTTPBadRequest(reason=f"Expected parameter {err}") from err
+
+
+@routes.get(VX + "/catalog/services/{service_key}/{service_version}/inputs")
+@login_required
+@permission_required("services.catalog.*")
+async def list_service_inputs_handler(request: Request):
+    with parameters_validation(request) as ctx:
+        # match, parse and validate
+        service_key: ServiceKey = request.match_info["service_key"]
+        service_version: ServiceVersion = request.match_info["service_version"]
+
+    # Evaluate and return validated model
+    response_model = await list_service_inputs(service_key, service_version, ctx)
+
+    # format response
+    enveloped: str = json.dumps(
+        {"data": [json.loads(m.json(by_alias=True)) for m in response_model]}
+    )
+
+    return web.Response(text=enveloped, content_type="application/json")
+
+
+@routes.get(VX + "/catalog/services/{service_key}/{service_version}/inputs/{input_key}")
+@login_required
+@permission_required("services.catalog.*")
+async def get_service_input_handler(request: Request):
+    with parameters_validation(request) as ctx:
+        # match, parse and validate
+        service_key: ServiceKey = request.match_info["service_key"]
+        service_version: ServiceVersion = request.match_info["service_version"]
+        input_key: ServiceInputKey = request.match_info["input_key"]
+
+    # Evaluate and return validated model
+    response_model = await get_service_input(
+        service_key, service_version, input_key, ctx
+    )
+
+    # format response
+    enveloped: str = json.dumps(
+        {"data": json.loads(response_model.json(by_alias=True))}
```

Comment:
a lot of dumps/loads/dumps... dict does not work?

Reply:
I agree this is a hack. I guess using the orjson approach we would not need this . The problem is that ``json`` might not know how to serialize some of the pydantic models produced with ``model_obj.dict()`` call . 

Let me see if I we have this dependency and will rewrite this

Reply:
changed dump by orjson